File: #condizionamento della somma.java
import java.util.Random;
import java.util.Scanner;
public class condizionamento_della_somma {
	public void condizionamento_della_somma() {
	//dati non perturbati
	Scanner scanner=new Scanner(System.in);
	System.out.println("inserire numero reale x:\n");
    double x=scanner.nextDouble();
    System.out.println("inserire numero reale y:\n");
    double y=scanner.nextDouble();
    //genero gli errori
    Random rand=new Random();
    double errx=rand.nextDouble()*Math.pow(10,-7);
    double erry=rand.nextDouble()*Math.pow(10,-7);
    //dati perturbati
    double xpt=x+errx;
    double ypt=y+erry;
    //somma perturbata e non perturbata
    double s=x+y;
    double st=xpt+ypt;
    double erx=Math.abs(x-xpt)/Math.abs(x);  //errore relativo x
    double ery=Math.abs(y-ypt)/Math.abs(y);  //errore relativo y
    double erst=Math.abs(s-st)/Math.abs(s);  //errore relativo somma
    //stampa risultati
    System.out.println("x="+x);
    System.out.println("y="+y);
    System.out.println("x perturbato="+xpt);
    System.out.println("y perturbato="+ypt);
    System.out.println("somma non perturbata="+s);
    System.out.println("somma perturbata="+st);
    System.out.println("errore relativo x="+erx);
    System.out.println("errore relativo y="+ery);
    System.out.println("errore relativo somma="+erst);
	
	
	

}}

File: #condizionamento della somma.py
#condizionamento della somma 
from random import random
print("inserire numero reale x:")
x=float(input())  #dati non perturbati
print("inserire numero reale y:")
y=float(input())
errx=random()*1.0e-7       #genero gli errori
erry=random()*1.0e-7
xpt=x+errx          #dati perturbati
ypt=y+erry

s=x+y            #somma non perturbata
st=xpt+ypt       #somma perturbata
erx=abs(x-xpt)/abs(x)     #errore relativo x
ery=abs(y-ypt)/abs(y)     #errore relativo y
erst=abs(s-st)/abs(s)     #errore relativo somma 
# stampa risultati
print("x="+str(x))
print("y="+str(y))
print("x perturbato="+str(xpt))
print("y perturbato="+str(ypt))
print("somma non perturbata="+str(s))
print("somma perturbata="+str(st))
print("errore relativo x="+str(erx))
print("errore relativo y="+str(ery))
print("errore relativo somma="+str(erst))


